applications/MappingApplication/custom_modelers/iga_fem_mapping_geometries_modeler.cpp

```diff
@@ -0,0 +1,242 @@
+
+        // Loop over all quadrature points in the interface model part to detect the different brep curves on surface
+        IndexType old_brep_curve_on_surface_id = 0;
+        for (auto cond_it = rInterfaceModelPart.ConditionsBegin(); cond_it != rInterfaceModelPart.ConditionsEnd(); ++cond_it){
+            // Get the quadrature point geometry
+            const GeometryPointerType p_geometry = cond_it->pGetGeometry();
+
+            // Get the parent geometry of the quadrature point (brep curve on surface)
+            const GeometryType& brep_curve_on_surface_geometry = p_geometry->GetGeometryParent(0);
+
+            // Get the id of the new brep curve on surface 
+            IndexType new_brep_curve_on_surface_id = brep_curve_on_surface_geometry.Id();
+
+            if (new_brep_curve_on_surface_id != old_brep_curve_on_surface_id){
+                for (auto geometry_itr = root_mp.GeometriesBegin(); geometry_itr != root_mp.GeometriesEnd(); geometry_itr ++){
+                    if (geometry_itr->Id() == new_brep_curve_on_surface_id){
+                        p_brep_curve_on_surface_vector.push_back(root_mp.pGetGeometry(geometry_itr->Id()));
+                    }
+                }
+                old_brep_curve_on_surface_id = new_brep_curve_on_surface_id;
+            }
+        }
+
+        // Determine next condition number
+        IndexType condition_id = (root_mp.NumberOfConditions() == 0)
+            ? 1 : (root_mp.ConditionsEnd() - 1)->Id() + 1;
+
+        for (IndexType i = 0; i < p_brep_curve_on_surface_vector.size(); i++){
+            coupling_conditions.CreateNewCondition("BrepCurveOnSurfaceCondition", condition_id, p_brep_curve_on_surface_vector[i], root_mp.ConditionsBegin()->pGetProperties(), 0);
+            condition_id += 1;
+        }
+    }
+
+    // In this method, nurbs_curve conditions (linear) are created on the fem interface (one curve per line element). This is used later to compute the intersection with the IGA boundary
+    void IgaFEMMappingGeometriesModeler::CreateFEMInterfaceNurbsCurveConditions(ModelPart& rInterfaceModelPart)
+    {
+        rInterfaceModelPart.CreateSubModelPart("coupling_conditions");
+        ModelPart& coupling_conditions = rInterfaceModelPart.GetSubModelPart("coupling_conditions");
+        const ModelPart& root_mp = rInterfaceModelPart.GetRootModelPart();
+
+        IndexType interface_node_id;
+        IndexType trial_interface_node_id;
+        IndexType trial_geom_node_id;
+
+        // Determine next condition number
+        IndexType condition_id = (root_mp.NumberOfConditions() == 0)
+            ? 1 : (root_mp.ConditionsEnd() - 1)->Id() + 1;
+
+        for (size_t node_index = 0; node_index < rInterfaceModelPart.NumberOfNodes() - 1; ++node_index)
+        {
+            interface_node_id = (rInterfaceModelPart.NodesBegin() + node_index)->Id();
+            std::vector< GeometryPointerType> p_geom_vec;
+            for (auto& ele_it: root_mp.Elements())
```

Comment:
```suggestion
            for (auto& r_elem: root_mp.Elements())
```
These are references not iterators

Reply:
You should change the rest of the code will not even compile

Reply:
Done